chore(release): add release notes for 2.31.0

armory-astrolabe opened the pull request:
Jira [CDSH-532]

deploy previews
https://deploy-preview-2247--armory-docs.netlify.app/continuous-deployment/release-notes/rn-armory-spinnaker/
 https://deploy-preview-2247--armory-docs.netlify.app/continuous-deployment/release-notes/rn-armory-spinnaker/armoryspinnaker_v2-31-0/

[CDSH-532]: https://armory.atlassian.net/browse/CDSH-532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

armory-astrolabe committed:
chore(release): add release notes for 2.31.0

mknous committed:
Update armoryspinnaker_v2-31-0.md

Work in progress PR (DO NOT MERGE YET) - need to refine some of the sections as I continue to add feature information.

mknous committed:
Update armoryspinnaker_v2-31-0.md

Added all the known feature flags to this version and updated the services with the feature work broken down by service. What still needs to be done:
- Review of breaking changes and known issues
- Some feature flags are missing info on how to turn them on/off

mknous committed:
Update armoryspinnaker_v2-31-0.md

Third update.
- Plugin compatibility should be updated
- 2 new feature flags need info on how to enable
- General review of highlighted updates
- Cameron will review feature flags to make sure they are up date.

mknous committed:
Update armoryspinnaker_v2-31-0.md

Tweaked Pipeline Triggers FF to include the PR which explains more about enabling.

Option to disable healthcheck for Google provider still needs some information.

aimeeu commented:
@mknous FYI since we are moving from 2.30 to 2.31, I need to version the docs site and update the version choice drop-down. I will make those updates in this PR.

aimeeu committed:
Update armoryspinnaker_v2-31-0.md

Change OSS Spinnaker to Spinnaker

aimeeu committed:
add Continuous Deployment to title

aimeeu committed:
add that this is a feature release

aimeeu committed:
Merge branch 'master' into release-notes-2.31.0-1701793837

aimeeu committed:
copy edits

aimeeu committed:
update config.toml for 2.31 and 2.30 versions

aimeeu committed:
added LTS to 2.20.5 and 2.28.7

aimeeu committed:
change release notes template heading to These are the three latest...

dogonthehorizon committed:
refact(rn): ensure most four recent release notes are shown (#2255)

We'll likely want to look at this deeper after 2.31, this is kinda quick
and hacky.

aimeeu committed:
remove 2.26 from previous version drop down
